frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -67,41 +72,94 @@ export class Gallery extends React.Component {
     addAllResourceI18n(i18n, this.state.config.translation, this.state.loggedUser.lang)
     i18n.changeLanguage(this.state.loggedUser.lang)
 
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.SHOW_APP(this.state.config.slug), handler: this.handleShowApp },
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+
+    props.registerLiveMessageHandlerList([
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.MODIFIED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrModifiedOrUndeleted },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.DELETED, optionalSubType: TLM_ST.FILE, handler: this.handleContentDeleted },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.UNDELETED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrModifiedOrUndeleted }
+    ])
+
     this.lightboxRotation = new LightboxRotation()
+  }
 
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+  // TLM Handlers
+  handleContentCreatedOrModifiedOrUndeleted = data => {
+    // A picture has been added or modified - let's reload the gallery list.
+    // The picture could have been added anywhere in the list.
+    // Any attempt to patch the image list may lead to inconsistency between a
+    // page reload and an update using data in the TLM.
+    this.loadGalleryList(this.state.config.appConfig.workspaceId, this.state.folderId)
   }
 
-  customEventReducer = ({ detail: { type, data } }) => {
-    const { state, props } = this
+  handleContentDeleted = data => {
+    const { state } = this
 
-    switch (type) {
-      case CUSTOM_EVENT.SHOW_APP(state.config.slug): {
-        console.log('%c<Gallery> Custom event', 'color: #28a745', type, data)
-        const newFolderId = qs.parse(data.config.history.location.search).folder_ids
-        if (data.config.appConfig.workspaceId !== state.config.appConfig.workspaceId || newFolderId !== state.folderId) {
-          this.setState({
-            config: data.config,
-            folderId: newFolderId
-          })
+    let newDisplayedPictureIndex = state.displayedPictureIndex
+    let newImagesPreviews = state.imagesPreviews
+
+    // We remove it from the list of previews
+    for (let i = 0; i < state.imagesPreviews.length; i++) {
+      if (state.imagesPreviews[i].contentId === data.content.content_id) {
+        // We found the picture to delete
+
+        // We remove it in the new list of images
+        newImagesPreviews = state.imagesPreviews.slice()
+        newImagesPreviews.splice(i, 1)
```

Comment:
I removed my use of splice. I used it because I wanted to avoid looping twice on the same array by using filter, but I found another solution. splice modified an array that belonged to the previous state, which is not good I guess.